extensions/amp-story/1.0/amp-story.js

```diff
@@ -2237,13 +2240,32 @@ export class AmpStory extends AMP.BaseElement {
 
     const pagesByDistance = this.getPagesByDistance_();
 
-    this.mutateElement(() => {
+    const preloadAllPages = () => {
       pagesByDistance.forEach((pageIds, distance) => {
         pageIds.forEach((pageId) => {
           const page = this.getPageById(pageId);
           page.setDistance(distance);
         });
       });
+    };
+
+    this.mutateElement(() => {
+      if (
+        !isExperimentOn(this.win, 'amp-story-load-first-page-only') ||
+        !prioritizeActivePage
+      ) {
+        return preloadAllPages();
+      }
+
+      const activePageId = devAssert(pagesByDistance[0][0]);
+      new Promise((res, rej) => {
+        const page = this.getPageById(activePageId);
+        page.setDistance(0);
+        page.signals().whenSignal(CommonSignals.LOAD_END).then(res);
+        page.signals().whenSignal(EventType.SWITCH_PAGE).then(rej);
```

Comment:
This is pretty smart and crafty, but this builds on top of a system that every single eng on this team would love to refactor eventually. Can we rely on either the store service, or another call to `preloadPagesByDistance_` (I know you're trying to avoid storing a promise in a ivar) instead of these events?